mcmcfit.py

```diff
@@ -1,871 +1,393 @@
-
-    # Read in GP params using fromString function from mcmc_utils.py
-    ampin_gp = Param.fromString('ampin_gp', input_dict['ampin_gp'])
-    ampout_gp = Param.fromString('ampout_gp', input_dict['ampout_gp'])
-    tau_gp = Param.fromString('tau_gp', input_dict['tau_gp'])
-
-    # Read in file names containing eclipse data, as well as output plot names
-    files = []
-    output_plots = []
-    for ecl in range(0,neclipses):
-        files.append(input_dict['file_{0}'.format(ecl)])
-        output_plots.append(input_dict['plot_{0}'.format(ecl)])
-
-
-    # Output file code - needs completing
-    # This file contains the data and best fit.
-    # The first three columns are the data (x, y and y error)
-    # The next column is the CV flux
-    # The next columns are the flux from wd, bright spot, disc and donor
-    # Setup header for output file format
-    # outfile_header =
-
-    # Create a model from the first eclipses (eclipse 0) parameters
-    parNames = ['wdFlux_0', 'dFlux_0', 'sFlux_0', 'rsFlux_0', 'q', 'dphi',\
-        'rdisc_0', 'ulimb_0', 'rwd', 'scale_0', 'az_0', 'fis_0', 'dexp_0', 'phi0_0']
-    if complex:
-        parNames.extend(['exp1_0', 'exp2_0', 'tilt_0', 'yaw_0'])
-    # List of values obtained from input file using fromString function from mcmc_utils.py
-    parList = [Param.fromString(name, input_dict[name]) for name in parNames]
-    #parList = [Param.fromString(name, input_dict[name]) for name in parNames if Param.isValid]
-    # If fitting using GPs use GPLCModel, else use LCModel
-    if useGP:
-        model = GPLCModel(parList,complex,ampin_gp,ampout_gp,tau_gp)
-    else:
-        model = LCModel(parList,complex)
-
-    # pickle is used for parallelisation
-    # pickle cannot pickle methods of classes, so we wrap
-    # the ln_prior, ln_like and ln_prob functions here to
-    # make something that can be pickled
-    def ln_prior(parList):
-        model.pars = parList
-        return model.ln_prior()
-    def ln_like(parList,phi,y,e,width=None):
-        model.pars = parList
-        #return model.ln_like(phi,y,e,width)
-        lnlike = model.ln_like(phi,y,e,width)
-        if np.isfinite(lnlike):
-            return lnlike
+    comp_scat = bool(int(input_dict['comp_scat']))
+
+    # neclipses no longer strictly necessary, but can be used to limit the
+    # maximum number of fitted eclipses
+    try:
+        neclipses = int(input_dict['neclipses'])
+    except KeyError:
+        neclipses = -1
+
+    if use_gp:
+        # TODO: Impliment the GP version of the Eclipses.
+        # Read in GP params using fromString function from mcmc_utils.py
+        ampin_gp = Param.fromString('ampin_gp', input_dict['ampin_gp'])
+        ampout_gp = Param.fromString('ampout_gp', input_dict['ampout_gp'])
+        tau_gp = Param.fromString('tau_gp', input_dict['tau_gp'])
+
+    # Start by creating the overall Model. Gather the parameters:
+    core_par_names = ['rwd', 'dphi', 'q']
```

Comment:
Neat little trick - you already have these on the class, since they are class-level attributes. Using the class attribute is better since if you change them you only have to change in one place.

```python
core_par_names = LCModel.node_par_names
```

Reply:
That is pretty nifty! Used this in the GP too:
 `node_par_names = Eclipse.node_par_names + ('ampin_gp', 'ampout_gp', 'tau_gp')`